skrf/io/touchstone.py

```diff
@@ -91,6 +92,19 @@ def n_ansys_impedance_values(self) -> int:
         return self.rank * 2
 
     @property
+    def rank(self) -> int:
+        return self._rank
+
+    @rank.setter
+    def rank(self, x: int) -> None:
+        self._rank = x
+
+        # If the rank changes, 'numbers_per_line' needs to be recalculated.
+        if "numbers_per_line" in self.__dict__:
```

Comment:
I'm not sure if this is even needed, the rank should definitely not change while parsing

Reply:
Thank you for your professional suggestion.
The `rank` property was mainly set to provide an entry point for **removing the `numbers_per_line` cache**. However, since you mentioned that the **rank should definitely not change**, the need to remove the cache becomes meaningless.
I will remove this part of the modification.